lib/connect.js

```diff
@@ -260,9 +261,8 @@ function connect(token, options) {
     } else {
       log.info('Error while connecting to a Room:', error);
     }
+    throw error;
   });
-
-  return cancelableRoomPromise;
```

Comment:
@makarandp0 Can you revert this change? We need to return a `CancelablePromise` here. Your change will return a regular `Promise` I think.

Reply:
Good point about `CancelablePromise`, I will revert this change, We can take/discuss this separately.